### PR TITLE
[4.0] Text Filters - accessibility etc

### PR DESCRIPTION
**Pull request by brianteeman**
We have a large block of text on the text filter tab at the top and at the bottom. Among the many problems with a large block of text like this is accessibility. Text like "see notes at the bottom" just doesn't work for assistive tech. Luckily there is a really easy solution. Using the html elements `<summary>` and `<details>`. The `<details>` element generates a simple no-JavaScript widget to show/hide element contents, optionally by clicking on its child `<summary>` element. It is fully accessible as the summary element is presented as a button to the accessibility tree.

Notes 
1. the text can be improved but its beyond the scope of this PR
2. it is expected and desired behaviour that opening a second detail does not close the first

### Testing instructions
npm i is needed "only" to check my bad scss ;) You can still test with patchtester you just wont get all the styling

### Expected result
filters

### CODE REVIEW

File: layouts/joomla/content/text_filters.php [hunk 1]
```diff
@@ -6,15 +6,31 @@
  * @copyright   Copyright (C) 2005 - 2019 Open Source Matters, Inc. All rights reserved.
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
+use Joomla\CMS\Language\Text;
```
Comment (Quy) [inline layouts/joomla/content/text_filters.php]: Move after `defined`.

File: layouts/joomla/content/text_filters.php [hunk 2]
```diff
@@ -6,15 +6,31 @@
...
-		<p><?php echo $displayData->description; ?></p>
-	<?php endif; ?>
+	<details>
+		<summary class="filter-notes"><?php echo Text::_('COM_CONFIG_TEXT_FILTERS_SUMMARY'); ?></summary>
+		<div class="filter-notes"><?php echo Text::_('COM_CONFIG_TEXT_FILTERS_DESC'); ?></div>
+	</details>
+	<details>
+		<summary class="filter-notes"><?php echo Text::_('JGLOBAL_FILTER_TYPE_LABEL');?></summary>
+		<div class="filter-notes"><?php echo Text::_('JGLOBAL_FILTER_TYPE_DESC'); ?></div>
+</details>
```
Comment (Quy) [inline layouts/joomla/content/text_filters.php]: Indent. Repeat below.

File: layouts/joomla/content/text_filters.php [hunk 3]
```diff
@@ -8,13 +8,29 @@
...
 	<legend><?php echo $displayData->name; ?></legend>
-	<?php if (!empty($displayData->description)) : ?>
-		<p><?php echo $displayData->description; ?></p>
-	<?php endif; ?>
+	<details>
+		<summary class="filter-notes"><?php echo Text::_('COM_CONFIG_TEXT_FILTERS_SUMMARY'); ?></summary>
+		<div class="filter-notes"><?php echo Text::_('COM_CONFIG_TEXT_FILTERS_DESC'); ?></div>
+	</details>
+	<details>
+		<summary class="filter-notes"><?php echo Text::_('JGLOBAL_FILTER_TYPE_LABEL');?></summary>
```
Comment (Quy) [inline layouts/joomla/content/text_filters.php]:
```suggestion
		<summary class="filter-notes"><?php echo Text::_('JGLOBAL_FILTER_TYPE_LABEL'); ?></summary>
```

File: layouts/joomla/content/text_filters.php [hunk 4]
```diff
@@ -8,13 +8,29 @@
...
+		<div class="filter-notes"><?php echo Text::_('JGLOBAL_FILTER_TYPE_DESC'); ?></div>
+	</details>
+	<details>
+		<summary class="filter-notes"><?php echo Text::_('JGLOBAL_FILTER_TAGS_LABEL');?></summary>
```
Comment (Quy) [inline layouts/joomla/content/text_filters.php]:
```suggestion
		<summary class="filter-notes"><?php echo Text::_('JGLOBAL_FILTER_TAGS_LABEL'); ?></summary>
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 82dcea98fc12a86b5f140d6e91286df6cd8bd5cb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25394">issues.joomla.org/tracker/joomla-cms/25394</a>.</sub>

**Comment by Quy** [comment]
Instead of 4 boxes which pushes the table further down, can they all be in 1 box in `Expand for notes...` and still be accessible?

**Comment by brianteeman** [comment]
Yes they can be - but then we have the big wall of text again. No one ever reads big walls of text. If you want to propose better text that is not a manual then go for it - I dont have the energy for it.

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 82dcea98fc12a86b5f140d6e91286df6cd8bd5cb<br /><br />> it is expected and desired behaviour that opening a second detail does not close the first

To be honest: I like this.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25394">issues.joomla.org/tracker/joomla-cms/25394</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/25394">issues.joomla.org/tracker/joomla-cms/25394</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**